src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -68,7 +68,11 @@ const useFetchChannelHook = () => {
 
     try {
       const channelList = ChannelList.fromChannelAPI(channel, type[channelType]);
-      await channelList.saveIfLatest(localDb);
+      if (channel?.type === 'topics') {
+        await channelList.save(localDb);
+      } else {
+        await channelList.saveIfLatest(localDb);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a conditional check for the channel type before saving the `channelList`. This is a good approach if different types of channels require different handling. However, it's important to ensure that all possible channel types are accounted for. If there are more types than just 'topics', you should consider adding an `else if` clause or a default case to handle them.

Also, error handling could be improved here. Currently, if an error occurs, it's logged to the console but not handled in any other way. Depending on the application's requirements, you might want to display an error message to the user, retry the operation, or take some other action when an error occurs.

Here's an example of how you might modify this code:

```diff
     try {
       const channelList = ChannelList.fromChannelAPI(channel, type[channelType]);
       if (channel?.type === 'topics') {
         await channelList.save(localDb);
+      } else if (channel?.type === 'anotherType') {
+        // Handle another type of channel
       } else {
         await channelList.saveIfLatest(localDb);
       }
       refresh('channelList');
     } catch (e) {
-      console.log('error on helperChannelPromiseBuilder');
+      console.error('error on helperChannelPromiseBuilder', e);
+      // Add additional error handling here
     }
```
This change adds a new `else if` clause to handle another type of channel and improves the error logging by including the error object itself. It also includes a comment where you can add additional error handling as needed.


<!-- This is an auto-generated comment by OSS CodeRabbit -->